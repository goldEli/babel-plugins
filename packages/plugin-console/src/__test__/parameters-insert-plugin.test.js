const pluginTester = require("babel-plugin-tester").default;

pluginTester({
  plugin: require("../plugin/parameters-insert-plugin"),
  babelOptions: {
    parserOpts: {
      sourceType: "unambiguous",
      plugins: ["jsx"],
    },
  },
  tests: {
    "console.xx前插入CallExpression的AST": {
      code: `
			console.log(1);
        
			function func() {
					console.info(2);
			}
	
			export default class Clazz {
					say() {
							console.debug(3);
					}
					render() {
							return <div>{console.error(4)}</div>
					}
			}
			`,
      snapshot: true,
    },
  },
});
