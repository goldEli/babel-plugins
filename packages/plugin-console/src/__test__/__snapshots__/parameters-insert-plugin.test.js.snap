// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin console.xx前插入CallExpression的AST: console.xx前插入CallExpression的AST 1`] = `

console.log(1);
     
function func() {
		console.info(2);
}
	
export default class Clazz {
		say() {
				console.debug(3);
		}
		render() {
				return <div>{console.error(4)}</div>
		}
}

      ↓ ↓ ↓ ↓ ↓ ↓

console.log("unkown filename: (1, 0)");
console.log(1);

function func() {
  console.log("unkown filename: (4, 2)");
  console.info(2);
}

export default class Clazz {
  say() {
    console.log("unkown filename: (9, 4)");
    console.debug(3);
  }

  render() {
    return (
      <div>{[console.log("unkown filename: (12, 17)"), console.error(4)]}</div>
    );
  }
}


`;
